### ISSUE TITLE
Monorepo: Activate declarationMap Compiler Option

### ISSUE THREAD

**Issue by holgerd77** [ethereumjs/ethereumjs-monorepo#1965]
There was a suggestion from @MicahZoltu to activate the [declarationMap](https://www.typescriptlang.org/tsconfig#declarationMap) (`declarationMap=true`) TypeScript compiler option which seems to make a lot of sense, see also the associated meta issue #1946 

This should be added to our base TypeScript configuration file (`config/tsconfig.json`) and also actively tested (maybe with screenshot in VSCode) if properly working (before case would also make sense to post).

This task can be done before the Beta 1 or Beta 2 releases.

**Comment by paulmillr** [comment]
Keep in mind that including source maps will likely drastically increase the size of output builds. Some testing should be done before a PR is merged. `npm pack --dry-run` helps to understand package sizes.

**Comment by holgerd77** [comment]
Have given this a short local test by adding the directive to a local `tsconfig.prod.json` file in the `Block` package.

So this is adding a `d.ts.map` file - e.g. `block.d.ts.map` for all files, added files are pretty small and the build size addition is neglectable, e.g. unpacked 280 KB -> 287 JB, so I think we should really do, I actually want to have this feature myself for an enhanced VSCode experience. 😄 

Will nevertheless remove the `urgent` label, this is not even breaking, but since the task is so tiny we can or can not add on the side line.

If someone wants to pick up as a side task feel free (but please still in separate PR).
